ocaml/xapi/xapi_session.ml

```diff
@@ -711,15 +711,35 @@ let login_with_password ~__context ~uname ~pwd ~version ~originator =
   let pwd = Bytes.of_string pwd in
   wipe_params_after_fn [pwd] (fun () ->
       (* !!! Do something with the version number *)
-      if Context.preauth ~__context then
+      match Context.preauth ~__context with
+      | Some socket_kind when socket_kind = `root ->
```

Comment:
the `when` is redundant here you should be able to write:`Some 'root`, and that will also properly check exhaustiveness.